uk.co.compucorp.civicrm.tasksassignments/js/test/services/task_test.js

```diff
@@ -0,0 +1,225 @@
+        });
+      });
+
+      describe("when calling TaskService.sendReminder()", function() {
+        beforeEach(function() {
+          $httpBackend.whenPOST(/action=sendreminder&entity=Task/).respond(function(method, url, data, headers, params) {
+            request.headers = headers;
+            request.data = data;
+
+            return [200];
+          });
+
+          TaskService.sendReminder(1, TaskMock.reminderNote);
+          $httpBackend.flush();
+        });
+
+        it('calls to save() with Reminder Note', function() {
+          var params = {
+            json: {
+              notes: TaskMock.reminderNote
+            }
+          };
+          requestBody = Task.save.calls.argsFor(0)[1],
+
+          expect(requestBody.json).toEqual(params.json);
+        });
+      });
+    });
+  });
+
+  describe('TaskService', function() {
+    var Task, TaskService, $httpBackend, request = {};;
+
+    beforeEach(module('civitasks.appDashboard'));
+
+    beforeEach(inject(function(_TaskService_, _$httpBackend_) {
+      TaskService = _TaskService_;
+      $httpBackend = _$httpBackend_;
+    }));
+
+    beforeEach(function() {
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+      $httpBackend.whenGET(/action=get&entity=Task/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+    });
+
+    describe("save()", function() {
+      beforeEach(function() {
+        $httpBackend.whenPOST(/action=create&entity=Task/).respond(function() {
+          return [200, TaskMock.response.onSave];
+        });
+      });
+
+      it('responds correct Task details', function (){
+        TaskService.save(TaskMock.task).then(function(data){
+          expect(data).toEqual(TaskMock.task);
+        });
+
+        $httpBackend.flush();
+      })
+    });
+
+    describe("assign()", function() {
+      beforeEach(function() {
+        $httpBackend.whenPOST(/action=copy_to_assignment&entity=Task/).respond(function() {
+          return [200, TaskMock.response.onAssign];
+        });
+      });
+
+      it('responds assigned Task List', function (){
+        TaskService.assign(TaskMock.taskList, 1).then(function(data){
+          expect(data).toEqual(TaskMock.taskList);
+        });
+
+        $httpBackend.flush();
+      });
+    });
+
+    describe("saveMultiple()", function() {
+      beforeEach(function() {
+        $httpBackend.whenPOST(/action=create_multiple&entity=Task/).respond(function() {
+          return [200, TaskMock.response.onSaveMultiple];
+        });
+      });
+
+      it('responds saved Task List', function (){
+        TaskService.saveMultiple(TaskMock.taskList).then(function(data){
+          expect(data).toEqual(TaskMock.response.onSaveMultiple.values);
```

Comment:
you are manually setting up the backend to respond with `TaskMock.response.onSaveMultiple`
```js
beforeEach(function() {
  $httpBackend.whenPOST(/action=create_multiple&entity=Task/).respond(function() {
    return [200, TaskMock.response.onSaveMultiple];
  });
});
```
so how can 
```js
expect(data).toEqual(TaskMock.response.onSaveMultiple.values);
```
ever fails? You are manually setting the test to always respond with that data, this test cannot ever fail, hence it's not a meaningful test